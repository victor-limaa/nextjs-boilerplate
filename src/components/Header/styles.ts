import styled from 'styled-components'

export const Wrapper = styled.main`
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: flex;
  align-items: center;
  justify-content: center;
`

export const Title = styled.h2``
