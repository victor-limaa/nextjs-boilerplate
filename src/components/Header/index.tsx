import * as S from './styles'

const Header = ({ title = 'Welcome to Next.js Boilerplate!' }) => (
  <S.Wrapper>
    <S.Title>{title}</S.Title>
  </S.Wrapper>
)

export default Header
