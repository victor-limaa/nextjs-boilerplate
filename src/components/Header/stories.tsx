import { Story, Meta } from '@storybook/react/types-6-0'
import Header from '.'

export default {
  title: 'Header',
  component: Header,
  args: {
    title: 'Global Arg'
  }
} as Meta

export const Basic: Story = (args) => <Header {...args} />
export const GlobalArg: Story = (args) => <Header {...args} />

// Basic.args = {
//   title: 'Title vindo do args'
// }
