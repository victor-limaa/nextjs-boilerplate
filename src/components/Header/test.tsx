import { screen, render } from '@testing-library/react'

import Header from '.'

describe('<Header />', () => {
  it('It should render the heading', () => {
    const { container } = render(<Header />)

    expect(
      screen.getByRole('heading', { name: /next.js boilerplate/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('It should render color correctly', () => {
    const { container } = render(<Header />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092b' })
  })
})
