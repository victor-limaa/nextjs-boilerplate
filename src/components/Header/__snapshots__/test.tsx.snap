// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> It should render the heading 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

<main
  class="c0"
>
  <h1
    class=""
  >
    Welcome to Next.js Boilerplate!
  </h1>
</main>
`
